api/app/jobs/rdps_sfms.py

```diff
@@ -0,0 +1,181 @@
+                        if downloaded:
+                            self.files_downloaded += 1
+                            file_name = self.get_file_name_from_url(url)
+                            file_path = os.path.join(temporary_path, file_name)
+                            key = self.generate_s3_key(model_run_hour, weather_param, file_name)
+                            # If we've downloaded the file ok, we can now save it to S3 storage.
+                            try:
+                                async with get_client() as (client, bucket):
+                                    await client.put_object(Bucket=bucket, Key=key, Body=open(file_path, "rb"))
+                                    create_saved_model_run_for_sfms_url(self.session, url)
+                                    print(url)
+                                # self.grib_processor.process_grib_file(downloaded, model_info, self.session)
+                                # # Flag the file as processed
+                                # flag_file_as_processed(url, self.session)
+                            finally:
+                                # delete the file when done.
+                                os.remove(downloaded)
+            except Exception as exception:
+                self.exception_count += 1
+                # We catch and log exceptions, but keep trying to download.
+                # We intentionally catch a broad exception, as we want to try and download as much
+                # as we can.
+                logger.error("unexpected exception processing %s", url, exc_info=exception)
+
+    async def process_model_run(self, model_run_hour: int):
+        """Process a particular RDPS model run"""
+        logger.info(f"Processing RDPS model run {model_run_hour}Z")
+        for key, value in GRIB_LAYERS.items():
+            urls = list(get_regional_model_run_download_urls(self.now, model_run_hour, [value], MAX_MODEL_RUN_HOUR))
+            await self.process_model_run_urls(model_run_hour, key, urls)
+
+    async def process(self):
+        """Entry point for downloading and processing RDPS weather model grib files"""
+        for hour in get_model_run_hours_to_process():
+            try:
+                await self.process_model_run(hour)
+                if self.files_downloaded > 0:
+                    create_model_run_for_sfms(self.session, ModelEnum.RDPS, self.now, hour)
+            except Exception as exception:
+                # We catch and log exceptions, but keep trying to process.
+                # We intentionally catch a broad exception, as we want to try to process as much as we can.
+                self.exception_count += 1
+                logger.error("unexpected exception processing %s model run %d", self.model_type, hour, exc_info=exception)
+
+
+class RdpsJob:
+    async def _run_rdp_job(self):
```

Comment:
```suggestion
    async def run(self):
```
I guess this is a public method and can just be called `run` since it's defined for the `RdpsJob` class

Reply:
https://github.com/bcgov/wps/pull/3741/commits/dcc01d0f2a1f395ff42ae3275aafee8b26e4cb13